src/allocators/OnChainAllocator.sol

```diff
@@ -241,6 +243,46 @@ contract OnChainAllocator is IOnChainAllocator, Utility {
         return commitments;
     }
 
+    /// @inheritdoc IOnChainAllocator
+    function permit2Allocation(
+        address arbiter,
+        address depositor,
+        uint256 expires,
+        ISignatureTransfer.TokenPermissions[] calldata permitted,
+        DepositDetails calldata details,
+        bytes32 claimHash,
+        string calldata witness,
+        bytes32 witnessHash,
+        bytes calldata signature
+    ) external returns (Lock[] memory commitments) {
+        if (expires > type(uint32).max) {
+            revert InvalidExpiration(expires, type(uint32).max);
+        }
+
+        commitments = AL.permit2Allocation(
+            arbiter, depositor, expires, permitted, details, claimHash, witness, witnessHash, signature
+        );
+
+        // Allocate the claim
+        for (uint256 i = 0; i < commitments.length; i++) {
+            // Check the amount fits in the supported range
+            if (commitments[i].amount > type(uint224).max) {
+                revert InvalidAmount(commitments[i].amount);
+            }
+
+            _storeAllocation(
```

Comment:
since allocations are scoped to specific claim hashes anyway, I wonder if it'd be possible to store a single hash of all the respective commitments rather than one at a time

Reply:
this is more of a musing than a suggestion at this stage, i think we could explore that optimization down the line but just food for thought for now